src/dataMigration.js

```diff
@@ -73,6 +73,27 @@ const dataMigrations = [
           }
         });
       });
+
+      const nameResults = database.objects('Name').filtered('id = $0',
+                                settings.get(SETTINGS_KEYS.SUPPLYING_STORE_NAME_ID));
+      if (nameResults.length < 1) throw new Error('Supplying Store Name ID missing from settings');
+      const mainSupplyingStoreName = nameResults[0];
+
+      const requisitions = database.objects('Requisition')
+                                   .filtered('otherStoreName = Null').snapshot();
+
+      database.write(() => {
+        requisitions.forEach(requisition => {
+          if (!requisition.isRequest) return;
```

Comment:
I'm happy with first answer, but second answer is a bit hazy. They types are `// imprest, forecast, request or response`, that doesn't have anything to do with statuses. If a store somehow has a requisition of type 'imprest' or 'forecast', it'll run this migration code.